pkg/server/proto/server.proto

```diff
@@ -32,6 +32,11 @@ service Waypoint {
   // OIDC provider.
   rpc CompleteOIDCAuth(CompleteOIDCAuthRequest) returns (CompleteOIDCAuthResponse);
 
+  // Attempts to run a trigger given a trigger ID reference. If the trigger does
+  // not exist, we return not found. If the trigger exists but requires authentication
+  // we return an error.
+  rpc NoAuthRunTrigger(RunTriggerRequest) returns (RunTriggerResponse);
```

Comment:
Is there a reason you did this versus overloading this on `RunTrigger`?

Reply:
I think so... we have other APIs now that behave similarly I believe (esp with the Runner stuff).

Reply:
Sounds good, I'll combine them!

Reply:
We chatted about this again in slack, and turns out we can do this a better way. I've opened an issue to track the work for now: https://github.com/hashicorp/waypoint/issues/2990. And for this PR I added some code comments on the two funcs with a link to the issue!